Assets/MixedRealityToolkit/Utilities/Editor/Preferences/ProjectPreferences.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
+{
+    /// <summary>
+    /// Utility to save preferences that should be saved per project (i.e to source control) across MRTK. Supports primitive preferences bool, int, and float
+    /// </summary>
+    public class ProjectPreferences : ScriptableObject
+    {
+        // Dictionary is not Serializable by default and furthermore System.object is not Serializable
+        // Thus, it is difficult to create a generic data bag. Instead we will create instances for each key preference types
+        [System.Serializable]
+        private class BoolPreferences : SerializableDictionary<string, bool> { }
+
+        [System.Serializable]
+        private class IntPreferences : SerializableDictionary<string, int> { }
+
+        [System.Serializable]
+        private class FloatPreferences : SerializableDictionary<string, float> { }
+
+        [System.Serializable]
+        private class StringPreferences : SerializableDictionary<string, string> { }
+
+        [SerializeField]
+        private BoolPreferences boolPreferences = new BoolPreferences();
+
+        [SerializeField]
+        private IntPreferences intPreferences = new IntPreferences();
+
+        [SerializeField]
+        private FloatPreferences floatPreferences = new FloatPreferences();
+
+        [SerializeField]
+        private StringPreferences stringPreferences = new StringPreferences();
+
+        private const string FILE_NAME = "ProjectPreferences.asset";
+
+        private const MixedRealityToolkitModuleType MODULE = MixedRealityToolkitModuleType.Generated;
+
+        private static ProjectPreferences _instance;
+        private static ProjectPreferences Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    string filePath = MixedRealityToolkitFiles.MapRelativeFilePath(MODULE, FILE_NAME);
+                    if (string.IsNullOrEmpty(filePath))
+                    {
+                        // MapRelativeFilePath returned null, need to build path ourselves
+                        filePath = MixedRealityToolkitFiles.MapModulePath(MODULE) + "/" + FILE_NAME;
```

Comment:
Does this mean that now MRTK will have a file "ProjectPreferences.asset" under the MRTK.Generated folder when people use MRTK? Have we thought about what our guidance will be for upgrading between MRTK versions, if people have this asset? Will we ask people to delete the MRTK.Generated folder?

Reply:
No people should not be deleting this folder because they will lose all of their personal assets. I don't believe we have ever recommended upgrading this folder. Generally all of the profiles go here and anything else would be totally lost